src/loaders/compute_tools/microtrait/microtrait.py

```diff
@@ -3,23 +3,68 @@
 """
 import json
 import os
+import uuid
 from pathlib import Path
 
 import pandas as pd
 from rpy2 import robjects
 
+from src.common.product_models.field_names import (
+    FIELD_HEATMAP_KBASE_ID,
+    FIELD_HEATMAP_VALUES,
+    FIELD_HEATMAP_ROW_CELLS,
+    FIELD_HEATMAP_CELL_ID,
+    FIELD_HEATMAP_COL_ID,
+    FIELD_HEATMAP_CELL_VALUE)
 from src.loaders.common import loader_common_names
 from src.loaders.compute_tools.tool_common import (
-    FatalTuple, 
+    FatalTuple,
     ToolRunner,
     write_fatal_tuples_to_dict,
 )
+from src.loaders.compute_tools.tool_result_parser import (
+    create_jsonl_files,
+    MICROTRAIT_META,
+    MICROTRAIT_CELLS,
+    MICROTRAIT_DATA)
 
 # the name of the component used for extracting traits from microtrait's 'extract.traits' result
 TRAIT_COUNTS_ATGRANULARITY = 'trait_counts_atgranularity3'
 _GENE_NAME_COL = 'hmm_name'  # column name from the genes_detected_table file that contains the gene name
 _GENE_SCORE_COL = 'gene_score'  # column name from the genes_detected_table file that contains the gene score
 
+# The following features will be extracted from the MicroTrait result file as heatmap data
+_MICROTRAIT_TRAIT_DISPLAYNAME_SHORT = 'microtrait_trait-displaynameshort'  # used as column name of the trait
+_MICROTRAIT_TRAIT_DISPLAYNAME_LONG = 'microtrait_trait-displaynamelong'  # used as description of the trait
+_MICROTRAIT_TRAIT_VALUE = 'microtrait_trait-value'  # value of the trait (can be integer or 0/1 as boolean)
+_MICROTRAIT_TRAIT_TYPE = 'microtrait_trait-type'  # type of trait (count or binary)
+_MICROTRAIT_TRAIT_ORDER = 'microtrait_trait-displayorder'  # order of the trait defined by the granularity table used as the index of trait
+
+# The following features are used to create the heatmap metadata and rows
+_SYS_TRAIT_INDEX = 'trait_index'  # index of the trait
+_SYS_TRAIT_NAME = 'trait_name'  # name of the trait
+_SYS_TRAIT_DESCRIPTION = 'trait_description'  # description of the trait
+_SYS_TRAIT_CATEGORY = 'trait_category'  # category of the trait
+_SYS_TRAIT_VALUE = 'trait_value'  # value of the trait
+_SYS_TRAIT_TYPE = 'trait_type'  # value of the trait
+
+_SYS_DEFAULT_TRAIT_VALUE = 0  # default value (0 or False) for a trait if the value is missing/not available
+
+# The map between the MicroTrait trait names and the corresponding system trait names
+# Use the microtrait_trait-name column as the unique identifier for a trait globally,
+# the microtrait_trait-displaynameshort column as the column name,
+# microtrait_trait-displaynamelong column as the column description, and
+# microtrait_trait-value as the cell value
+_MICROTRAIT_TO_SYS_TRAIT_MAP = {
+    loader_common_names.MICROTRAIT_TRAIT_NAME: loader_common_names.SYS_TRAIT_ID,
```

Comment:
Can  these and the one below be moved from loader_common_names now?

Reply:
👍

Reply:
I was saying move the fields *from* loader common, since they're not common anymore, rather than *to* loader common

Reply:
:+1: